### PR TITLE
Fixed Progressbar Binding Error (ProgressBar Value is Not Nullable)

### PR DESCRIPTION
**Pull request by Sparrkle**
When the item in the ResultListBox was added, fixed progressbar binding error.

```
System.Windows.Data Error: 40 : BindingExpression path error: 'ProgressBarVisibility' property not found on 'object' ''Result' (HashCode=1859780700)'. BindingExpression:Path=Result.ProgressBarVisibility; DataItem='ResultViewModel' (HashCode=1859780700); target element is 'ProgressBar' (Name=''); target property is 'Visibility' (type 'Visibility')
```

### CODE REVIEW

File: Flow.Launcher/Themes/Base.xaml [hunk 1]
```diff
@@ -89,11 +91,7 @@
         <Setter Property="Minimum" Value="0" />
         <Setter Property="Visibility" Value="Visible" />
         <Setter Property="Foreground" Value="#26a0da " />
-        <Style.Triggers>
-            <DataTrigger Binding="{Binding Result.ProgressBar}" Value="{x:Null}">
-                <Setter Property="Visibility" Value="Collapsed" />
-            </DataTrigger>
-        </Style.Triggers>
+        <Setter Property="Visibility" Value="{Binding Result.IsProgressBarVisible, Converter={StaticResource BooleanToVisibilityConverter}}"/>
```
Comment (taooceros) [inline Flow.Launcher/Themes/Base.xaml:94]: Please help me understand why we want to put the Visibility binding to the theme file not directly in the Listbox xaml.

File: Flow.Launcher.Plugin/Result.cs [hunk 2]
```diff
@@ -201,7 +201,12 @@ public ValueTask<bool> ExecuteAsync(ActionContext context)
         /// <summary>
         /// Progress bar display. Providing an int value between 0-100 will trigger the progress bar to be displayed on the result
         /// </summary>
-        public int? ProgressBar { get; set; }
+        public int ProgressBarValue { get; set; }
+
+        /// <summary>
+        /// Progress bar visibility Check
+        /// </summary>
+        public bool IsProgressBarVisible { get; set; } = false;
```
[the remark this answers is not in the document]
Comment (Sparrkle) [reply Flow.Launcher.Plugin/Result.cs:209]: hmm,, I feel a little strange, too. but there are also items that dont use the progress bar.
I think we need to create an ```Result``` interface and separate it.

File: Flow.Launcher/Themes/Base.xaml (hunk 1 above)
Comment (Sparrkle) [reply Flow.Launcher/Themes/Base.xaml:94]: original code is in base.xaml. so i keeped it that way.
But I think we can move the Visibility binding to ```ResultListBox.xaml```.

Comment (taooceros) [reply Flow.Launcher/Themes/Base.xaml:94]: @onesounds do you have special reason for doing this? I didn't notice this when reviewing that pr.

Comment (taooceros) [reply Flow.Launcher/Themes/Base.xaml:94]: ok @Sparrkle could you please move the visibility binding to `ResultListBox.xaml`?

Comment (Sparrkle) [reply Flow.Launcher/Themes/Base.xaml:94]: Okay. i have moved it.

File: Flow.Launcher.Plugin/Result.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (taooceros) [reply Flow.Launcher.Plugin/Result.cs:209]: Sorry I didn't see this message. What do you think about `Nullable<int>` (i.e. i`nt?`)?

[the remark this answers is not in the document]
Comment (taooceros) [reply Flow.Launcher.Plugin/Result.cs:209]: > I think we need to create an Result interface and separate it.

Hmm I am not sure why we need an interface for Result? To make sure everything is properly initialized? If so, we can try the required property after upgrading to .Net 7

[the remark this answers is not in the document]
Comment (Sparrkle) [reply Flow.Launcher.Plugin/Result.cs:209]: the problem is value of the progress bar does not support ```Nullable<int>```.
```
System.Windows.Data Error: 6 : 'SystemConvertConverter' converter failed to convert value '<null>' (type '<null>'); fallback value will be used, if available. BindingExpression:Path=Result.ProgressBar; DataItem='ResultViewModel' (HashCode=-536391821); target element is 'ProgressBar' (Name=''); target property is 'Value' (type 'Double') InvalidCastException:'System.InvalidCastException: Null object cannot be converted to a value type.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at MS.Internal.Data.SystemConvertConverter.Convert(Object o, Type type, Object parameter, CultureInfo culture)
   at System.Windows.Data.BindingExpression.ConvertHelper(IValueConverter converter, Object value, Type targetType, Object parameter, CultureInfo culture)'
```

[the remark this answers is not in the document]
Comment (taooceros) [reply Flow.Launcher.Plugin/Result.cs:209]: oh ok....I see what you mean. What about creating a property in the viewmodel that is int and default 0? Then we can still use the nullable int for Result.cs.

[the remark this answers is not in the document]
Comment (Sparrkle) [reply Flow.Launcher.Plugin/Result.cs]: Oh, I understand. I just modified it!

File: Flow.Launcher/ResultListBox.xaml [hunk 3]
```diff
@@ -114,8 +114,18 @@
...
+                                <ProgressBar.Style>
+                                    <Style TargetType="ProgressBar" BasedOn="{StaticResource ProgressBarResult}">
+                                        <Setter Property="Visibility" Value="Visible"/>
+                                        <Style.Triggers>
+                                            <DataTrigger Binding="{Binding Result.ProgressBar}" Value="{x:Null}">
+                                                <Setter Property="Visibility" Value="Collapsed"/>
+                                            </DataTrigger>
+                                        </Style.Triggers>
+                                    </Style>
+                                </ProgressBar.Style>
```
Comment (taooceros) [inline Flow.Launcher/ResultListBox.xaml:127]: That's impressive！Thank you!

**Review by taooceros — APPROVED** [review]

**Pull request merged**